src/caa-cases/containers/caa-cases/caa-cases.component.ts

```diff
@@ -44,54 +45,102 @@ export class CaaCasesComponent implements OnInit {
   public errorMessages: ErrorMessage[];
   public noCasesFoundMessage = '';
 
+  @ViewChild('tabGroup') public tabGroup: MatTabGroup;
+
   constructor(private readonly store: Store<fromStore.CaaCasesState>,
               private readonly organisationStore: Store<fromOrganisationStore.OrganisationState>,
               private readonly userStore: Store<fromUserStore.UserState>,
               private readonly router: Router) {
-    // Identify whether user selected to view assigned cases or unassigned cases
-    this.caaCasesPageType = this.router && this.router.url && this.router.url.includes('unassigned-cases')
-      ? CaaCasesPageType.UnassignedCases
-      : CaaCasesPageType.AssignedCases;
+  }
+
+  public ngOnInit(): void {
+    // Set current page type
+    this.setCurrentPageType();
     // Set page title
+    this.setCurrentPageTitle();
+    // Set show hide filter button text
+    this.setShowHideFilterButtonText();
+    // Set filter type to "all-assignees" for assigned cases and "none" for unassigned cases
+    this.setSelectedFilterTypeAndValue();
+
+    // Load case types from store based on current page type
+    this.loadCaseTypes(this.selectedFilterType, this.selectedFilterValue);
+
+    // Get selected cases to share from store
+    this.shareCases$ = this.store.pipe(select(fromStore.getShareCaseListState));
+    this.shareCases$.subscribe(shareCases => this.selectedCases = converters.toSearchResultViewItemConverter(shareCases));
+
+    // Load selected organisation details from store
+    this.organisationStore.dispatch(new fromOrganisationStore.LoadOrganisation());
+    this.selectedOrganisation$ = this.organisationStore.pipe(select(fromOrganisationStore.getOrganisationSel));
+
+    // Load users of selected organisation from store
+    this.userStore.dispatch(new fromUserStore.LoadUsers(0));
+    this.selectedOrganisationUsers$ = this.userStore.pipe(select(fromUserStore.getGetUserList));
+
+    // Load cases based on page type and set table configuration
+    this.loadCasesAndSetTableConfig();
+  }
+
+  public loadCaseTypes(selectedFilterType: string, selectedFilterValue: string): void {
```

Comment:
@johnbenjamin-hmcts - I may have missed it but is there a unit test for this function, or is it covered indirectly?

Reply:
Thank you @Daniel-Lam I have now written couple of unit tests to cover this.

Reply:
Thanks for adding some unit tests, @johnbenjamin-hmcts.